TimeTracking/TimeTracking.php

```diff
@@ -20,239 +20,266 @@
+     * Prints the time tracking inputs within the bugnote-add form
+     * @param type $p_event
+     * @param type $p_bug_id
+     */
+    function ev_bugnote_add_form( $p_event, $p_bug_id ) {
+        $t_enabled_on_bugnote_add_form = plugin_config_get('enabled_on_bugnote_add_form');
+        if( $t_enabled_on_bugnote_add_form ) {
+            if( TimeTracking\user_can_edit_bug_id( $p_bug_id ) ) {
+                TimeTracking\print_bugnote_add_form( $p_bug_id );
+            }           
+        }
+    }
+
+    /**
+     * Validates time tracking submitted data when adding bugnotes
+     */
+    function ev_bugnote_add_validate( $p_event, $p_bugnote_text, $p_bug_id ) {
+        $t_time_imput = gpc_get_string( 'plugin_timetracking_time_input', '' );
+        if( !is_blank( $t_time_imput ) ) {
+            if( TimeTracking\user_can_edit_bug_id( $p_bug_id ) ) {
+                $t_parsed = TimeTracking\parse_gpc_time_record();
+            }
+        }
+        return $p_bugnote_text;
+    }
+
+    /**
+     * Creates a time tracking record from submitted data when adding bugnotes
+     */
+    function ev_bugnote_added( $p_event, $p_bug_id, $p_bugnote_id ) {
+        $t_time_imput = gpc_get_string( 'plugin_timetracking_time_input', '' );
+        if( !is_blank( $t_time_imput ) ) {
+            if( TimeTracking\user_can_edit_bug_id( $p_bug_id ) ) {
+                $t_record = TimeTracking\parse_gpc_time_record();
+                $t_record['bugnote_id'] = $p_bugnote_id;
+                $t_record['bug_id'] = $p_bug_id;
+                TimeTracking\create_record( $t_record );
+            }
+        }
+    }
+
+    function ev_view_bug_details( $p_event, $p_bug_id ) {
+        if( TimeTracking\user_can_view_bug_id( $p_bug_id ) ) {
+            $t_records = TimeTracking\get_records_for_bug( $p_bug_id );
+            if( $t_records ) {
+                TimeTracking\print_bug_details_row( $p_bug_id );
+            }
+        }
+    }
+
+    function ev_layout_content_begin( $p_event ) {
+        if( TimeTracking\stopwatch_enabled() && TimeTracking\stopwatch_exists() ) {
+            TimeTracking\print_stopwatch_header_control();
+        }
+    }
+
+    function ev_view_bug( $p_event, $p_bug_id ) {
+        if( TimeTracking\user_can_view_bug_id( $p_bug_id ) ) {
+            TimeTracking\print_bug_timetracking_section( $p_bug_id );
+        }
+    }
+
+    /**
+     *
+     */
+    static function getConfig() {
+        return array(
+            # old thresholds
+            /*
+            'admin_own_threshold'   => DEVELOPER,
+            'view_others_threshold' => MANAGER,
+            'admin_threshold'       => ADMINISTRATOR,
+            */
+            # new thresholds
+            'view_threshold'    => DEVELOPER,
+            'edit_threshold'    => DEVELOPER,
+            'reporting_threshold'   => MANAGER,
+
+            'stopwatch_enabled' => ON,
+
+            'categories'    => '',
+            'enabled_on_bugnote_add_form'    => ON
```

Comment:
maybe a shorter name is cleaner?
Eg: `bugnote_form_enabled`
(placing "enabled" at the end, following the other option naming)

Reply:
I've though about putting enabled at the end, but I was not able to find a good name. bugnote_form_enabled is not clear IMHO, I've given a look to original config option on standard MantisBT and name was  $g_time_tracking_without_note, if you think it will be better this name can be used, please let me know

Reply:
`time_tracking_without_note` was used in a different way: to allow a time entry with an empty bugnote. Because in core, a bugnote is required to have content, that options was used to override that validation and allow inserting an empty bugnote for the purpose of storing the time entry.
Here we don't have that limitation, and not use for that option.

As we are talking about the naming, and following your latest change, i'm fine with just `bugnote_form_enabled`.